src/measure_theory/lp_space.lean

```diff
@@ -94,4 +108,82 @@ by simp [snorm, hp0_lt]
 
 end zero
 
+@[simp] lemma snorm_neg {f : α → F} : snorm (-f) p μ = snorm f p μ :=
+by simp [snorm]
+
+
+section borel_space
+variable [borel_space E]
+
+lemma mem_ℒp.neg_mem_ℒp {f : α → E} (hf : mem_ℒp f p μ) : mem_ℒp (-f) p μ :=
+⟨measurable.neg hf.1, by simp [hf.right]⟩
+
+
+variable [topological_space.second_countable_topology E]
+
+lemma rpow_nnnorm_add_le_const_mul_add_rpow_nnnorm {β} {f g : β → F} (hp1 : 1 ≤ p) (a : β) :
+  (nnnorm (f a + g a)) ^ p ≤ 2 ^ (p - 1) * ((nnnorm (f a)) ^ p + (nnnorm (g a)) ^ p):=
```

Comment:
I am completely confused by this lemma. Why do you state it for functions, instead of for fixed elements of a vector space? Also, are you aware that we already have some convexity inequalities on nnreals in the file `analysis/mean_inequalities.lean` (see for instance `nnreal.rpow_arith_mean_le_arith_mean_rpow` there)?

Reply:
Indeed there is no reason to have functions there. (f a) and (g a) should simply be some x and y.
Thanks for the nnreal convexity inequalities.
That lemma needs of a big rewrite.